bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieImpl.java

```diff
@@ -714,12 +717,14 @@ public BookieImpl(ServerConfiguration conf, StatsLogger statsLogger,
                 this.stateManager.transitionToReadOnlyMode();
             }
         }
-
+        this.journalExistCountDown = new CountDownLatch(1);
         // instantiate the journals
         journals = Lists.newArrayList();
         for (int i = 0; i < journalDirectories.size(); i++) {
-            journals.add(new Journal(i, journalDirectories.get(i),
-                    conf, ledgerDirsManager, statsLogger.scope(JOURNAL_SCOPE), allocator));
+            Journal journal = new Journal(i, journalDirectories.get(i),
+                    conf, ledgerDirsManager, statsLogger.scope(JOURNAL_SCOPE), allocator);
+            journal.setThreadAliveCounterDown(journalExistCountDown);
```

Comment:
Better for dependencies to be injected through the constructor. Tests use one constructor, which doesn't need the latch, and the BookieImpl uses the other. So you won't need to update lots of tests.

Reply:
Actually, if we decide whether a bookie thread should exit by the CountDownLatch,  many behaviours in the unit test are broken even if `Journal` has two constructors. 
For lots of tests, the `Journal` does not have the CountDownLatch,  shutdown the `BookieImpl` or stop the `BookieServer` will be blocked.